src/shared/sleep-config.c

```diff
@@ -108,77 +111,187 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+/* Get the list of battery */
+static int battery_enumerator_new(sd_device_enumerator **ret) {
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        int r;
+
+        assert(ret);
+
+        r = sd_device_enumerator_new(&e);
+        if (r < 0)
+                return log_oom();
+
+        r = sd_device_enumerator_add_match_subsystem(e, "power_supply", true);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_add_match_property(e, "POWER_SUPPLY_TYPE", "Battery");
+        if (r < 0)
+                return r;
+
+        *ret = TAKE_PTR(e);
+
+        return 0;
+}
```

Comment:
in our coding style, functions should either log about all errors, or about none (in which case it's up the caller to log or handle the error). This function logs in one error path via log_oom(), but not in the others.

usually functions that are "library-like" should not log on their own, and functions that are "main-program-like" should. But the lines are blurry and exceptions are OK. Hence, make up your mind what kind of function this should be, and then either add log calls on all error paths, or remove them from all and ensure you return an error properly.

here, i'd probably remove the log_oom() call up top, and just propagate `return r;`

Reply:
yes. will take care of it.